software/main.py

```diff
@@ -0,0 +1,266 @@
+import image_processor
+import camera
+import cv2
+import time
+from math import *
+import numpy as np
+import movement
+import calculations
+import communication
+
+def Orbit(radius, xcord, reso_x_mid,dist):
+    speedY = 0
+    speedX = 0.5
+    speedR = 1000*speedX/(640-radius)
+
+    if radius > (dist-5):
+        speedY += (radius-dist)/100
+    elif radius > (dist+5):
+        speedY += (radius-dist)/100
+    
+    if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+        speedR += (reso_x_mid- xcord) / 100
+
+    return [speedX, speedR, speedY]
+
+
+def main_loop():
+    tdist=[100, 150, 200, 250, 300]
+    tsize=[79,53,40,32,28]
+    tspeeds=[550, 650, 725, 800, 900]
+    state="find_ball"
+    spin = 10
+    debug=True
+    basket_color="magenta"
+    basket_height = 0.5
+    cam=camera.RealsenseCamera(exposure=100) #defaulti peal on depth_enabled = True
+    processor = image_processor.ImageProcessor(cam, debug=debug, color_config = "colors/colors.pkl")
+    robot=movement.OmniRobot()
+    reso_x_mid=424
+    calculator = calculations.Calculations()
+    comms=communication.Communication()
+    processor.start()
+    start=time.time()
+    fps=0
+    frame=0
+    frame_cnt=0
+    radius=250
+    ball_side=0
+    calibration_speed=0.15
+    
+    try:
+        while True:
+            if state == "throw":
+                processedData = processor.process_frame(aligned_depth=True)
+            else:
+                processedData = processor.process_frame(aligned_depth=False)
+
+            frame_cnt+=1
+            frame+=1
+            if state == "try":
+                
+                robot.try_motors()
+                continue
+            if state == "tmotor":
+                spt=int(input("NNA TSPEED: "))
+                robot.test_thrower(spt)
+                continue
+            if state=="cameratest":
+                try:
+                    processedData = processor.process_frame(aligned_depth=True)
+                    processedData.balls.sort(key= lambda x: x.size)
+                    #targeted_ball=processedData.balls[-1]
+                    print(processedData.basket_m)
+                    
+                    
+                except:
+                    print("Ei leidnud")
+                    
+                
+            
+            if frame%30 == 0:
+                frame = 0
+                end = time.time()
+                fps = 30/(end-start)
+                start = end
+                #[Object: x=7; y=212; size=15.0; distance=212; exists=True]  
+                print(fps)  
+
+            if state == "find_ball":
+                print("--------Searching ball--------")
+                if len(processedData.balls)!=0:
+                    state="move"
+                else:
+                    if ball_side==1:

```

Comment:
This variable is very confusing. What does side mean? Is it on the left or right? If it is supposed to represent a boolean, I suggest using True or False

Reply:
Tracks if ball coordinate smaller or bigger than frames middle x coordinate.

Reply:
I suggest naming this variable so its meaning is clear, like `ball_left_side = True`, or similar. Then it defines what value it holds